mentor-match-app/src/components/profile/LoggedUserProfile.jsx

```diff
@@ -0,0 +1,490 @@
+import React, { useState, useEffect } from 'react'
+import {
+  Dialog,
+  DialogContent,
+  Button,
+  Typography,
+  CircularProgress,
+  Box,
+  Stack,
+  Divider,
+  Link,
+  IconButton,
+  Fab
+} from '@mui/material'
+import {
+  Business as BusinessIcon,
+  Language as LanguageIcon,
+  LocationOn as LocationIcon,
+  Link as LinkIcon,
+  SchoolRounded as SchoolIcon,
+  Person as PersonIcon,
+  Close as CloseIcon,
+  Edit as EditIcon
+} from '@mui/icons-material'
+import ProfilePicture from '../ProfilePicture'
+import { useUser } from '../../hooks/useUser'
+import { getFormAnswers } from '../../api/forms'
+import { useSnackbar } from 'notistack'
+import { Form, Formik } from 'formik'
+import * as yup from 'yup'
+import { LoadingButton } from '@mui/lab'
+import TextField from '../../components/questions/text/TextField'
+import { updateUserProfile } from '../../api/users'
+import SubmittedFormsSection from './SubmittedFormsSection'
+import { ref, uploadBytes, getDownloadURL } from 'firebase/storage'
+import { storage } from '../../api/firebaseConfig'
+import ProfileField from './ProfileField'
+
+const schema = yup.object().shape({
+  name: yup.string().required('Please enter your name'),
+  title: yup.string().required('Please enter your title'),
+  affiliation: yup.string().required('Please enter your affiliation'),
+  //profilePicture: yup.string().url('Invalid URL'),
+  websiteUrl: yup.string().url('Invalid URL'),
+  publicProfile: yup.boolean()
+})
+
+const LoggedUserProfile = ({ openDialog, setOpenDialog }) => {
+  const [loading, setLoading] = useState(true)
+  const { user, refreshUser } = useUser()
+  const [formAnswers, setFormAnswers] = useState({})
+  const [enableEdit, setEnableEdit] = useState(false)
+  const { enqueueSnackbar } = useSnackbar()
+
+  const onSubmit = async (values, { setSubmitting }) => {
+    setSubmitting(true)
+    try {
+      const res = await updateUserProfile(user, values)
+      if (!res.ok) {
+        setSubmitting(false)
+        return enqueueSnackbar(res.error, { variant: 'error' })
+      }
+      enqueueSnackbar('Profile updated successfully', { variant: 'success' })
+      await refreshUser()
+      setSubmitting(false)
+      setEnableEdit(false)
+      //handleDialogOpen()
+    } catch (error) {
+      console.error('Error during profile update:', error)
+      enqueueSnackbar('An error occurred during profile update', {
+        variant: 'error'
+      })
+      setSubmitting(false)
+    }
+  }
+
+  useEffect(() => {
+    if (openDialog) {
+      const fetchFormAnswers = async () => {
+        const ans = await getFormAnswers(user.uid)
+        setFormAnswers(ans)
+        console.log('formAnswers', ans)
+        setLoading(false)
+      }
+      fetchFormAnswers()
+    }
+  }, [openDialog, user.uid])
+
+  const initialValues = {
```

Comment:
Add Null Safety and Default Values

```suggestion
    const initialValues = {
-   name: user.displayName,
-   description: user.profileDescription,
-   affiliation: user.affiliation,
-   title: user.title,
-   location: user.location,
-   profilePicture: user.profilePicture,
-   profileDescription: user.profileDescription,
-   websiteUrl: user.websiteUrl
+   name: user?.displayName || '',
+   description: user?.profileDescription || '',
+   affiliation: user?.affiliation || '',
+   title: user?.title || '',
+   location: user?.location || '',
+   profilePicture: user?.profilePicture || '',
+   profileDescription: user?.profileDescription || '',
+   websiteUrl: user?.websiteUrl || ''
  }

+ if (!user) return null;
```